plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/DeltaLakeSchemaSupport.java

```diff
@@ -446,8 +469,11 @@ public static Map<String, Boolean> getColumnsNullability(MetadataEntry metadataE
         return getColumnProperties(metadataEntry, node -> node.get("nullable").asBoolean());
     }
 
-    public static Map<String, Boolean> getColumnIdentities(MetadataEntry metadataEntry)
+    public static Map<String, Boolean> getColumnIdentities(MetadataEntry metadataEntry, ProtocolEntry protocolEntry)
     {
+        if (protocolEntry.supportsWriterFeature() && !protocolEntry.writerFeaturesContains("identityColumns")) {
+            return ImmutableMap.of();
```

Comment:
Why we check for checkConstraints now and we didn't before?
If the previous logic was flawed, and this is a fix, it would be nice to have it as a separate commit.

Reply:
Separated into "Add ProtocolEntry to DeltaLakeTableHandle" commit.